thrust/device_malloc.h

```diff
@@ -44,27 +44,28 @@ THRUST_NAMESPACE_BEGIN
  *  #include <thrust/device_malloc.h>
  *  #include <thrust/device_free.h>
  *  ...
- *  // allocate some memory with device_malloc
+ *  // allocate some integers with device_malloc
  *  const int N = 100;
- *  thrust::device_ptr<void> void_ptr = thrust::device_malloc(N);
+ *  thrust::device_ptr<int> int_array = thrust::device_malloc<int>(N);
  *
- *  // manipulate memory
+ *  // manipulate integers
  *  ...
  *
  *  // deallocate with device_free
- *  thrust::device_free(void_ptr);
+ *  thrust::device_free(int_array);
  *  \endcode
  *
  *  \see device_ptr
  *  \see device_free
  */
-inline thrust::device_ptr<void> device_malloc(const std::size_t n);
+template<typename T>
+inline thrust::device_ptr<T> device_malloc(const std::size_t n);
 
-/*! This version of \p device_malloc allocates sequential device storage for
- *  new objects of the given type.
+/*! This version of \p device_malloc allocates sequential device storage
+ *  for bytes.
```

Comment:
Not sure if I'm a fan of the "for bytes" wording here...Maybe something like
"allocates untyped sequential device storage"? or generic instead of untyped?

Reply:
Good idea, thanks. I've updated the description to "allocates untyped sequential device storage".